source/grid/grid_in.cc

```diff
@@ -1731,6 +1731,8 @@ GridIn<dim, spacedim>::read_msh(std::istream &in)
   std::vector<CellData<dim>>                 cells;
   SubCellData                                subcelldata;
   std::map<unsigned int, types::boundary_id> boundary_ids_1d;
+  bool                                       is_hex_mesh = false;
+  bool                                       is_tet_mesh = false;
```

Comment:
Would you mind renaming `is_hex_mesh` to `is_quad_or_hex_mesh` and `is_tet_mesh` with `is_tria_or_tet_mesh`?